### ISSUE TITLE
Data loss when merging from two VictoriaMetrics servers

### ISSUE THREAD

**Issue by fatgit** [jacksontj/promxy#203]
I've setup Promxy over two VictoriaMetrics servers and Grafana looks at promxy.

I stopped VictoriaMetrics#1 server. Everything looks good, Grafana draws a full graph.

I started VictoriaMetrics#1 server again and result of the merge:

Data from VictoriaMetrics#1
<img width="1428" alt="vic-1" src="https://user-images.githubusercontent.com/11026137/62152147-ce831580-b2b6-11e9-8e96-aef63c021e61.png">

Data from VictoriaMetrics#2
<img width="1542" alt="vic-2" src="https://user-images.githubusercontent.com/11026137/62152165-d6db5080-b2b6-11e9-9d4e-013179af2f30.png">

Data from Promxy
<img width="1532" alt="proxmy" src="https://user-images.githubusercontent.com/11026137/62152102-ae535680-b2b6-11e9-9241-867a91cff610.png">

How to make that the result of the merge would be like from VictoriaMetrics#2 instead VictoriaMetrics#1?

My config:

 ```
promxy:
  server_groups:
    - static_configs:
        - targets:
          - vic-1:8428
          - vic-2:8428
      anti_affinity: 2s
      remote_read: false
      path_prefix: /
      http_client:
        dial_timeout: 1s
        tls_config:
          insecure_skip_verify: true
```

**Comment by jacksontj** [comment]
That is very odd, this is one of the most basic functionalities in promxy (and has pretty good test coverage). I just manually tested this same situation with some prometheus hosts as origin and am unable to reproduce. 

![Screenshot from 2019-08-05 12-56-15](https://user-images.githubusercontent.com/1619869/62491351-7cdbfe80-b780-11e9-83f6-4347945ad57d.png)


Can you upload (ideally both, at least one of ) trace logs and tcpdump during the queries?

**Comment by fatgit** [comment]
I've removed old cluster and setup a new one with next configuration:
![image](https://user-images.githubusercontent.com/11026137/62639512-838b8280-b8f4-11e9-91ee-0d375474488c.png)
So, I have two instances of promxy, one for two independent Prometheus and one for two independent victoriametrics

I stopped one by one prometheus serices and what I see
<img width="879" alt="Test victoria - Grafana 2019-08-07 09-15-16" src="https://user-images.githubusercontent.com/11026137/62639878-44116600-b8f5-11e9-8cfc-1a16e9fdd223.png">

No Prometheus service shutdown affects promxy of prometheus. It's perfect.

But when I stopped prometheus#1 proxmy of Victoriametrics lost data, at the same time as stopping prometheus#2 didn't affect proxmy.

**Comment by fatgit** [comment]
And If I go directly to promxy console of victoriametris, I can see data for this time:
![image](https://user-images.githubusercontent.com/11026137/62640616-d5350c80-b8f6-11e9-9208-e51ebdbcee30.png)
It seems that for some reason the problem is only in grafana which  does not return data
![image](https://user-images.githubusercontent.com/11026137/62640821-407ede80-b8f7-11e9-8c6d-00aa6ff99856.png)

**Comment by jacksontj** [comment]
Interesting from your report here it seems that promxy is fine with prometheus as the origin just not with VM as the origin. My assumption here is that VM is returning the wrong values during that time range. From looking at this most recent post it seems that VM is returning NaN during that "outage" window instead of nothing. So the next step would be comparing the results from prometheus and VM to promxy's query (since presumalby they are returning different values).

**Comment by fatgit** [comment]
Do you mean that?
Promxy above prometheus:
![image](https://user-images.githubusercontent.com/11026137/62902499-ab278400-bd14-11e9-932d-2000a7c51fe1.png)
above VM
![image](https://user-images.githubusercontent.com/11026137/62902535-c1354480-bd14-11e9-8f6f-79fc7159dcdd.png)

While both of prometheus work, the results are the same

**Comment by jacksontj** [comment]
Not quite. From the graphs above you had shown that a gap on Prometheus was
stitched together correctly by promxy. Whereas the VM metrics aren't
stitched together in promxy right -- which means that VM and prom are
returning different values, so ideally we'd get the output of both pairs
(the ones that should get stitched together).

**Comment by jacksontj** [comment]
I was able to make some time this evening to look into this and as I suspected this seems to be a VictoriaMetrics bug (https://github.com/VictoriaMetrics/VictoriaMetrics/issues/153). Once that is fixed in VM this should start working as expected.

**Comment by jacksontj** [comment]
Upstream VM has a fix that'll be in the next release, closing out this issue :)

**Comment by valyala** [comment]
The fix is available in VictoriaMetrics [v1.26.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases)

**Comment by fatgit** [comment]
![image](https://user-images.githubusercontent.com/11026137/63529768-1cf09200-c4ba-11e9-82c9-52b6d0e58c4c.png)
Looks nice!
